src/dscanner/analysis/lambda_return_check.d

```diff
@@ -52,15 +51,45 @@ extern (C++) class LambdaReturnCheck(AST) : BaseAnalyzerDmd
 		if (lambda.fbody.isReturnStatement() is null)
 			return;
 
-		auto tokenRange = tokens.filter!(t => t.loc.fileOffset > lambda.loc.fileOffset)
-			.filter!(t => t.loc.fileOffset < lambda.endloc.fileOffset)
-			.find!(t => t.value == TOK.goesTo);
+		auto lambdaRange = tokens.filter!(t => t.loc.fileOffset > lambda.loc.fileOffset)
+			.filter!(t => t.loc.fileOffset < lambda.endloc.fileOffset);
+		auto tokenRange = lambdaRange.find!(t => t.value == TOK.goesTo);
 
 		if (!tokenRange.canFind!(t => t.value == TOK.leftCurly))
 			return;
 
 		if (!tokenRange.canFind!(t => t.value == TOK.leftParenthesis))
-			addErrorMessage(cast(ulong) lambda.loc.linnum, cast(ulong) lambda.loc.charnum, KEY, MSG);
+		{
+			auto start = tokenRange.front.loc.fileOffset;
+			AutoFix fix0;
+			auto firstParam = (*(lambda.getParameterList().parameters))[0];
```

Comment:
Note that lambdas can have 0 parameters.

Reply:
Sure they can, but a 0 parameter lambda must have parens, right? If it has parens it won't enter that if branch

Reply:
so  what `(*(lambda.getParameterList().parameters))[0];` return in that case?

Reply:
Yes. I've tried a lambda with no parens and no params and it fails:
`b.map!( => { return a * a + 2; });`

```
tmp.d(5): Error: expression expected, not `=>`
tmp.d(5): Error: found `{` when expecting `)` following template argument list
tmp.d(5): Error: found `return` when expecting `;` following statement `b.map!0` on line tmp.d(5)
tmp.d(5): Error: declaration expected, not `)`
tmp.d(6): Error: unmatched closing brace
```

With parens for the empty param the compilation succeeds.

Reply:
No, a lambda like that won't enter that branch. There's also plenty of lambdas like that in phobos and the pipeline hasn't failed.